packages/core/core/src/BundleGraph.js

```diff
@@ -479,7 +518,14 @@ export default class BundleGraph {
       }
     }
 
-    this._graph.removeNode(bundleGroupNode);
+    // This function can be reentered through removeBundle above. In this case,
+    // the node may already been removed.
+    if (this._graph.hasContentKey(bundleGroupNode.id)) {
+      this._graph.removeNode(
+        this._graph.getNodeIdByContentKey(bundleGroupNode.id),
+      );
+    }
```

Comment:
Wouldn't this have thrown previously anyway? (Because of the assertion in `removeNode`)
Why is the condition needed now?

Reply:
I believe this check was added because the issue was being thrown, so this prevents the case of being thrown if the node was revisited by the removal chain.